x-pack/plugins/security_solution/public/common/containers/related_entities/related_hosts/index.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useMemo } from 'react';
+import type { inputsModel } from '../../../store';
+import type { InspectResponse } from '../../../../types';
+import { RelatedEntitiesQueries } from '../../../../../common/search_strategy/security_solution/related_entities';
+import type { RelatedHost } from '../../../../../common/search_strategy/security_solution/related_entities/related_hosts';
+import { useSearchStrategy } from '../../use_search_strategy';
+
+export interface UseUserRelatedHostsResult {
+  inspect: InspectResponse;
+  totalCount: number;
+  relatedHosts: RelatedHost[];
+  refetch: inputsModel.Refetch;
+  loading: boolean;
+}
+
+interface UseUserRelatedHostsParam {
+  userName: string;
+  indexNames: string[];
+  from: string;
+  skip?: boolean;
+}
+
+export const useUserRelatedHosts = ({
+  userName,
+  indexNames,
+  from,
+  skip = false,
+}: UseUserRelatedHostsParam): UseUserRelatedHostsResult => {
+  const {
+    loading,
+    result: response,
+    search,
+    refetch,
+    inspect,
+  } = useSearchStrategy<RelatedEntitiesQueries.relatedHosts>({
+    factoryQueryType: RelatedEntitiesQueries.relatedHosts,
+    initialResult: {
+      totalCount: 0,
+      relatedHosts: [],
+    },
+    errorMessage: 'error',
```

Comment:
Please consider replace this with more description or just leave it as undefined, so it'll apply the default message.